extensions/amp-story/0.1/animation-presets-utils.js

```diff
@@ -77,3 +77,39 @@ export function whooshIn(startX, startY, endX, endY) {
     },
   ];
 }
+
+/**
+ * Checks if page dimensions are larger than those of the target to be animated.
+ * @param {StoryAnimationDimsDef} dimensions Dimensions of page and target.
+ */
+export function pageIsLargerThanTarget(dimensions) {
+  return dimensions.pageWidth > dimensions.targetWidth ||
+         dimensions.pageHeight > dimensions.targetHeight;
+}
+
+/**
+ * Calculate target scaling factor so that it is at least 25% larger than the page.
+ * @param {StoryAnimationDimsDef} dimensions Dimensions of page and target.
+ * @return {number}
+ */
+export function calculateTargetScalingFactor(dimensions) {
+  const widthFactor = dimensions.pageWidth > dimensions.targetWidth ?
+                      dimensions.pageWidth / dimensions.targetWidth : 1;
+  const heightFactor = dimensions.pageHeight > dimensions.targetHeight ?
+                       dimensions.pageHeight / dimensions.targetHeight : 1;
+  return Math.max(widthFactor, heightFactor) + 0.25;
```

Comment:
Maybe instead of adding 0.25, we should multiply by 1.25?  If you need to increase something by 6000x for it to fit the viewport, increasing it by 6000.25 won't give you much space to pan.

Also, nit: whatever approach we take, that magic number should be a constant